controllers/operatorconfigmap_controller.go

```diff
@@ -317,10 +322,35 @@ func (c *ClusterVersionReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	c.log.Info("prometheus rules deployed", "prometheusRule", klog.KRef(prometheusRule.Namespace, prometheusRule.Name))
 
+	// deletion phase
+	if !instance.GetDeletionTimestamp().IsZero() {
```

Comment:
I don't see a finalizer on configmap, so I think we might get an event even if this cm was just recreated 🤔

Reply:
This is the user created configmap are we creating it now?

Reply:
Yes, we should be the ones to create an empty one. After this change, the config map becomes the root resource for this controller as it is tied to the namespace lifespan. 

But, even with that in mind, I don't see why so we need a finalizer. A finalizer is used if you must keep a resource alive to complete nay cleanup operation, and I don't believe this is the case here

Reply:
> Yes, we should be the ones to create an empty one
- should this be part of bundle?

> A finalizer is used if you must keep a resource alive to complete nay cleanup operation
- yes and at the same time we are tying the deletion of config map as deletion of csi deployments & ds
- it might so happen someone mistook the config map and deleted it and going by current logic we'll cleanup csi as well
- I know finaliser on configmap doesn't solve deletion of csi and my point only is, deletion and creation of this configmap deletes and creates csi w/o finalizer, are we aiming for it?

---

regarding configmap name:
- I hope all reviewers is aware that this configmap doesn't exist as of now and what is the suggestion on how to create it, via bundle? another PR?
- what we deploy as part of bundle is `ocs-client-operator-manager-config` cm ([ref](https://github.com/red-hat-storage/ocs-client-operator/blob/main/config/manager/kustomization.yaml#L10)) and do we want to create a new `ocs-client-operator-config` cm ([ref](https://github.com/red-hat-storage/ocs-client-operator/blob/main/controllers/clusterversion_controller.go#L54))